alerting/provider/awsses/awsses.go

```diff
@@ -0,0 +1,167 @@
+package awsses
+
+import (
+	"fmt"
+	"github.com/TwiN/gatus/v5/alerting/alert"
+	"github.com/TwiN/gatus/v5/core"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ses"
+	"strings"
+)
+
+const (
+	CharSet = "UTF-8"
+)
+
+// AlertProvider is the configuration necessary for sending an alert using PagerDuty
+type AlertProvider struct {
+	AccessKeyID     string `yaml:"access-key-id"`
+	SecretAccessKey string `yaml:"secret-access-key"`
+	Region          string `yaml:"region"`
+
+	From string `yaml:"from"`
+	To   string `yaml:"to"`
+
+	// DefaultAlert is the default alert configuration to use for endpoints with an alert of the appropriate type
+	DefaultAlert *alert.Alert `yaml:"default-alert,omitempty"`
+
+	// Overrides is a list of Override that may be prioritized over the default configuration
+	Overrides []Override `yaml:"overrides,omitempty"`
+}
+
+// Override is a case under which the default integration is overridden
+type Override struct {
+	Group string `yaml:"group"`
+	To    string `yaml:"to"`
+}
+
+// IsValid returns whether the provider's configuration is valid
+func (provider *AlertProvider) IsValid() bool {
+	registeredGroups := make(map[string]bool)
+	if provider.Overrides != nil {
+		for _, override := range provider.Overrides {
+			if isAlreadyRegistered := registeredGroups[override.Group]; isAlreadyRegistered || override.Group == "" || len(override.To) == 0 {
+				return false
+			}
+			registeredGroups[override.Group] = true
+		}
+	}
+
+	return len(provider.From) > 0 && len(provider.To) > 0 &&
+		((len(provider.AccessKeyID) == 0 && len(provider.SecretAccessKey) == 0) || (len(provider.AccessKeyID) > 0 && len(provider.SecretAccessKey) > 0))
```

Comment:
When would `(len(provider.AccessKeyID) == 0 && len(provider.SecretAccessKey) == 0)` be acceptable if `len(provider.From) > 0 && len(provider.To) > 0`?

Reply:
AWS access key and secret are optional variables if you use IAM authentication. So they both should either be empty or filled in.

Reply:
In that case, should the condition check for IAM auth if the keys aren't provided?

Reply:
IAM authentication is implicit on resources within AWS, and doesn't really need to be checked, if that's what you mean.

Reply:
Oh I see, I was interpretting it as the aws profile config but I suppose it's implicit if you deploy to aws.

Reply:
Can you add a comment that clarifies this?

i.e. `if both AccessKeyID and SecretAccessKey are specified, we'll use these to authenticate, otherwise if neither are specified, then we'll fall back on IAM auth